rocAL/rocAL/include/pipeline/tensor.h

```diff
@@ -101,8 +119,16 @@ class rocalTensorInfo {
         }
     }
     void set_tensor_layout(RocalTensorlayout layout) {
-        _layout = layout;
-        set_max_shape();
+        if(_layout != layout && _layout != RocalTensorlayout::NONE) {
+            RocalTensorlayout old_layout = _layout;
+            _layout = layout;
+            std::vector<size_t> new_dims(_num_of_dims, 0);
+            get_modified_dims_from_layout(old_layout, _layout, new_dims);
+            set_dims(new_dims);
+        } else {
+            _layout = layout;
+            set_max_shape();
```

Comment:
is set_max_shape() only meant to be used in the else condition ?

Reply:
Yes, for if condition it will be invoked by set_dims API